c_common/front_end_common_lib/include/spin1-wfi.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2020 The University of Manchester
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <stdint.h>
+
+//! \file
+//! \brief Declaration that is common but not in spin1_api.h
+
+/*! \brief Wait for interrupt.
+ *  \details Inline version of code that appears in spin1_api so that we can
+ *      get more compact code. For a description of what this actually does,
+ *      see <a href="https://developer.arm.com/
+documentation/ddi0311/d/system-control-coprocessor/cp15-register-descriptions/
+cp15-c7-core-control-operations">the relevant ARM documentation</a>
+ *      (it's hardware magic, specific to the ARM968).
+ */
+static inline void spin1_wfi(void) {
```

Comment:
I guess you want this inline; is it worth giving it a different name since it isn't in spin1_api?  It might make people look for it in the wrong place...

Reply:
The name is carried forward from previous uses. Suggestions welcome!

Reply:
Simple option! `wait_for_interrupt()`